src/background/transactions/TransactionService.ts

```diff
@@ -52,9 +67,88 @@ export class TransactionService {
 
   async initialize() {
     await this.transactionsStore.ready();
-    const pending = getPendingTransactions(this.transactionsStore.getState());
+    const transactions = this.transactionsStore.getState();
+    const pending = getPendingTransactions(transactions);
     this.transactionsPoller.add(pending.map(toPollingObj));
     this.addListeners();
+    if (transactions.length) {
+      this.startPurgeInterval({ leading: true });
+    }
+  }
+
+  private startPurgeInterval({ leading } = { leading: false }) {
+    if (leading) {
+      this.performPurgeCheck(); // make leading call
+    }
+    this.purgeInterval.start(ONE_DAY);
+  }
+
+  private schedulePurgeCheck = throttle(
+    () => {
+      this.performPurgeCheck();
+    },
+    FOUR_MINUTES,
+    { leading: false } // Invoke no sooner and no more frequent than FOUR_MINUTES
+  );
+
+  /**
+   * Purges from cache all transactions belonging to {address} in {chainId}
+   * which have a nonce less than or equals to {fromNonce}
+   */
+  private async purgeEntries({
+    address,
+    fromNonce,
+    chainId,
+  }: {
+    address: string;
+    fromNonce: number;
+    chainId: ChainId;
+  }) {
+    const transactions = await this.transactionsStore.getSavedState();
+    const candidates = transactions.filter((item) => {
+      return (
+        normalizeAddress(item.transaction.from) === normalizeAddress(address) &&
+        normalizeChainId(item.transaction.chainId) === chainId &&
+        item.transaction.nonce <= fromNonce
+      );
+    });
+    this.transactionsStore.bulkDeleteTransactionsByHash(
+      candidates.map((item) => item.hash)
+    );
+  }
+
+  /** Finds local transactions which our backend is aware of and removes them */
+  private async performPurgeCheck() {
+    const transactions = await this.transactionsStore.getSavedState();
+
+    type Address = string;
+    type Key = `${Address}:${ChainId}`;
+    const map = new Map<Key, { hash: string; timestamp: number }>();
+
+    for (const item of transactions) {
+      const chainId = normalizeChainId(item.transaction.chainId);
+      const key = `${item.transaction.from}:${chainId}` as const;
+      map.set(key, { hash: item.hash, timestamp: item.timestamp });
+    }
+
+    for (const [key, { hash, timestamp }] of map.entries()) {
+      const [address, chainIdStr] = key.split(':');
+      const chainId = chainIdStr as ChainId;
+      const network = await getNetworkByChainId(chainId);
```

Comment:
Why not using networksStore here?

Reply:
Ok then. This will just remove an extra call for a chain info probably

Reply:
But it seems that we want fresh data here